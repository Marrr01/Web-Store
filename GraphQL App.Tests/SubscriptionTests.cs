using CookieCrumble;
using HotChocolate.Execution;
using Xunit;

namespace GraphQL_App.Tests
{
    public class SubscriptionTests
    {
        [Fact]
        public async Task PasswordChanged_CorrectQuery_ReturnsTwoUsers()
        {
            using var cts = new CancellationTokenSource(2500);

            await using var result = await TestServices.ExecuteRequestAsync(b => b.SetQuery(
                """
                subscription t {
                  passwordChanged {
                    login
                    password
                  }
                }
                """
                ));

            var snapshot = new Snapshot();
            var count = 0;

            await foreach (var item in result.ExpectResponseStream()
                .ReadResultsAsync().WithCancellation(cts.Token))
            {
                snapshot.Add(item, $"Result {++count}");

                if (count == 2)
                {
                    break;
                }
            }

            snapshot.Match();
        }
    }
}
