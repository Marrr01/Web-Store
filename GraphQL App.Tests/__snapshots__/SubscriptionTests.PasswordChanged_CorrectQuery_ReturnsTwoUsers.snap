Result 1
---------------
{
  "data": {
    "passwordChanged": {
      "login": "someLogin 1",
      "password": "someNewPassword"
    }
  }
}
---------------

Result 2
---------------
{
  "data": {
    "passwordChanged": {
      "login": "someLogin 2",
      "password": "someNewPassword"
    }
  }
}
---------------
